### PR TITLE
Feature/add filing count for company

### PR DESCRIPTION
**Pull request by rosealti**

### CODE REVIEW

File: python/edgar_filing_searcher/parsers/main.py [hunk 1]
```diff
@@ -15,11 +16,20 @@ def create_url_list(url_edgar_current_events):
...
+def update_filing_counts(cik_no_list):
+    """This function counts the number of filings and adds it to the Company table"""
+    for cik_no in cik_no_list:
+        company_in_table = Company.query.filter_by(cik_no=cik_no).first()
+        filing_count = EdgarFiling.query.filter_by(cik_no=cik_no).count()
+        company_in_table.filing_count = filing_count
+        db.session.commit()
+
+
+def send_data_to_db(company_row, edgar_filing_row, data_13f_table_parameter):
```
Comment (datdo) [inline python/edgar_filing_searcher/parsers/main.py:28]: parameter? You should pretty much never include the variable type or designation in the name. Maybe data_13f_table_rows might be better.

File: python/tests/parsers/test_main.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+        """This function tests update_filing_counts"""
+        cik_no = '0001171592'
+
+        update_filing_counts([cik_no])
+
+        assert Company.query.filter_by(cik_no=cik_no).first().filing_count == 1
+
+    def test_send_data_to_db_savesCompanyInDb(self):
+        """This function tests send_data_to_db_savesCompanyInDb"""
+        self.company_2 = Company(cik_no="000984343", company_name="True Blue", filing_count=0)
```
Comment (datdo) [inline python/tests/parsers/test_main.py:92]: no need to make these variables fields of the class. Just leave them as local variables.

File: python/tests/parsers/test_main.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        send_data_to_db(self.company_2, self.edgar_filing_2, self.data_13f_table_2)
+
+        assert Company.query.filter_by(cik_no='000984343').first() == self.company_2
+
+    def test_send_data_to_db_savesEdgarFilingInDb(self):
+        """This function tests send_data_to_db_savesEdgarFilingInDb"""
+        self.company_3 = Company(cik_no="8673434", company_name="Purple Company", filing_count=1)
+        self.edgar_filing_3 = EdgarFiling(accession_no="3453456", cik_no="8673434",
+                                          filing_date=datetime.fromisoformat("2000-06-11"))
+        self.data_13f_table_3 = [Data13f(equity_holdings_id="67896567",
```
Comment (datdo) [inline python/tests/parsers/test_main.py:120]: same as above

File: python/tests/parsers/test_main.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+        mock_function.return_value = MagicMock(text=current_events_text)
+        fake_url = ""
+
+        actual = create_url_list(fake_url)
+
+        assert actual == [
+            'https://www.sec.gov/Archives/edgar/data/1850858/0001850858-21-000001-index.html',
+            'https://www.sec.gov/Archives/edgar/data/1852858/0001852858-21-000001-index.html',
+            'https://www.sec.gov/Archives/edgar/data/1835714/0001085146-21-001095-index.html',
+            'https://www.sec.gov/Archives/edgar/data/1446194/0001011712-21-000002-index.html']
```
Comment (TheNorthEestern) [inline python/tests/parsers/test_main.py:76]: Not a deal breaker.
```suggestion
        assert actual == [
            'https://www.sec.gov/Archives/edgar/data/1850858/0001850858-21-000001-index.html',
            'https://www.sec.gov/Archives/edgar/data/1852858/0001852858-21-000001-index.html',
            'https://www.sec.gov/Archives/edgar/data/1835714/0001085146-21-001095-index.html',
            'https://www.sec.gov/Archives/edgar/data/1446194/0001011712-21-000002-index.html'
        ]
```

File: python/tests/parsers/test_main.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+                                    cusip='00846U101',
+                                    value='22967078',
+                                    ssh_prnamt='180644',
+                                    ssh_prnamt_type='None',
+                                    put_call='None',
+                                    investment_discretion='SOLE',
+                                         other_manager='None',
+                                         voting_authority_sole='22967078',
+                                         voting_authority_shared='0',
+                                         voting_authority_none='0')]
```
Comment (TheNorthEestern) [inline python/tests/parsers/test_main.py:159]:
```suggestion
        data_13f_table_4 = [Data13f(equity_holdings_id="67896567",
                                    accession_no='78945835',
                                    cik_no='009039443',
                                    name_of_issuer='Agilent Technologies',
                                    title_of_class='COM',
                                    cusip='00846U101',
                                    value='22967078',
                                    ssh_prnamt='180644',
                                    ssh_prnamt_type='None',
                                    put_call='None',
                                    investment_discretion='SOLE',
                                    other_manager='None',
                                    voting_authority_sole='22967078',
                                    voting_authority_shared='0',
                                    voting_authority_none='0')]
```

File: python/tests/parsers/test_main.py [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
```
Comment (TheNorthEestern) [inline python/tests/parsers/test_main.py:109]:
```suggestion
        data_13f_table_2 = [Data13f(equity_holdings_id="67896567",
                                    accession_no='0001420506',
                                    cik_no='00054654983',
                                    name_of_issuer='Agilent Technologies',
                                    title_of_class='COM',
                                    cusip='00846U101',
                                    value='22967078',
                                    ssh_prnamt='180644',
                                    ssh_prnamt_type='None',
                                    put_call='None',
                                    investment_discretion='SOLE',
                                    other_manager='None',
                                    voting_authority_sole='22967078',
                                    voting_authority_shared='0',
                                    voting_authority_none='0')]
```

File: python/tests/parsers/test_main.py [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
```
Comment (TheNorthEestern) [inline python/tests/parsers/test_main.py:134]:
```suggestion
        data_13f_table_3 = [Data13f(equity_holdings_id="67896567",
                                    accession_no='3453456',
                                    cik_no='654656465',
                                    name_of_issuer='Agilent Technologies',
                                    title_of_class='COM',
                                    cusip='00846U101',
                                    value='22967078',
                                    ssh_prnamt='180644',
                                    ssh_prnamt_type='None',
                                    put_call='None',
                                    investment_discretion='SOLE',
                                    other_manager='None',
                                    voting_authority_sole='22967078',
                                    voting_authority_shared='0',
                                    voting_authority_none='0')]
```

File: python/edgar_filing_searcher/api/__init__.py [hunk 8]
```diff
@@ -5,11 +5,13 @@
 from edgar_filing_searcher.api.routes.company import company_blueprint
 
 
-def create_app():
+def create_app(obj=None):
     """Create Flask app and create database."""
     app = Flask(__name__)
     app.config['SQLALCHEMY_DATABASE_URI'] = os.environ.get('DB_CONNECTION_STRING')
     app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+    if obj:
```
Comment (TheNorthEestern) [inline python/edgar_filing_searcher/api/__init__.py:13]: What is `obj` here?

**Review by TheNorthEestern — APPROVED** [review]
LGTM 🚀

**Pull request merged**